shared/video-sdk/authentication-workflow/project-implementation/cpp.mdx

```diff
@@ -100,33 +119,27 @@
     with:
 
     ```cpp
-    if (token == "")
-    {
-        // Fetch new token 
-        token = fetchToken(serverUrl, channelName, tokenRole, uid, expireTime);
-        if (token == "")
-        {
-            AfxMessageBox(L"Invalid Token.");
-            return;
-        }
-    }
-    if (0 == agoraEngine->joinChannel(token.c_str(), channelName.c_str(), NULL, uid))
-    {
-        isJoin = true;
-    }
+    CString channel;
+	GetDlgItemText(IDC_EDIT1, channel);
+	channelName = std::string(CT2CA(channel));
+	if(channel == "")
+	{
+		AfxMessageBox(L"Please write a channel name in the text box");
+		return;
+	}
+	else
+	{
+		// Fetch new token 
+		token = fetchToken(serverUrl, channelName, tokenRole, uid, expireTime);
+
+		if (0 == agoraEngine->joinChannel(token.c_str(), channelName.c_str(), NULL, uid))
+		{
+			isJoin = true;
+		}
+	}
     ```
     This examples uses a different `joinchannel()` call to support a uid other than 0.
 
-6. **Set `token` as an empty string in the `Leave` button event listener:**
```

Comment:
It is important to set `token` as empty string in Leave button listener. Else when a user leave the channel and join after some time then as I am calling fetchToken() only in case of token blank(Please see my implementation). So in that user will go for joining with may be expired token. Again **we have not handled the case when token is expired**  . Only the case here when token is about to expire and we need to fetch new token and join the channel. @billy-the-fish  Please clear on this.